backend/internal/handlers/classrooms/assignments/assignments.go

```diff
@@ -321,3 +321,122 @@ func (s *AssignmentService) getAssignmentRubric() fiber.Handler {
 		})
 	}
 }
+
+func (s *AssignmentService) getGradedCount() fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		// Parse assignmentID
+		assignmentID, err := strconv.ParseInt(c.Params("assignment_id"), 10, 64)
+		if err != nil {
+			return errs.BadRequest(err)
+		}
+
+		// Query work status counts
+		counts, err := s.store.CountWorksByState(c.Context(), int(assignmentID))
+		if err != nil {
+			return errs.InternalServerError()
+		}
+
+		// Count graded/ungraded works
+		gradedWorks := 0
+		ungradedWorks := 0
+		for state, count := range counts {
+			if state == models.WorkStateGradingCompleted || state == models.WorkStateGradePublished {
+				gradedWorks += count
+			} else {
+				ungradedWorks += count
+			}
+		}
+
+		return c.Status(http.StatusOK).JSON(fiber.Map{
+			"assignment_id": assignmentID,
+			"status": fiber.Map{
+				"graded":   gradedWorks,
+				"ungraded": ungradedWorks,
+			},
+		})
+	}
+}
+
+func (s *AssignmentService) getAssignmentStatus() fiber.Handler {
+	return func(c *fiber.Ctx) error {
+		// Parse assignmentID and classroomID
+		assignmentID, err := strconv.ParseInt(c.Params("assignment_id"), 10, 64)
+		if err != nil {
+			return errs.BadRequest(err)
+		}
+		classroomID, err := strconv.ParseInt(c.Params("classroom_id"), 10, 64)
+		if err != nil {
+			return errs.BadRequest(err)
+		}
+
+		// Query work status counts
+		counts, err := s.store.CountWorksByState(c.Context(), int(assignmentID))
+		if err != nil {
+			return errs.InternalServerError()
+		}
+
+		// Count assignment statuses
+		acceptedWork := counts[models.WorkStateAccepted]
+		startedWork := counts[models.WorkStateStarted]
+		submittedWork := counts[models.WorkStateSubmitted]
+		workInGrading := counts[models.WorkStateGradingAssigned] +
+			counts[models.WorkStateGradingCompleted] +
+			counts[models.WorkStateGradePublished]
+
+		// Determine unaccepted works using number of students in classroom
+		numStudents, err := s.store.GetNumberOfUsersInClassroom(c.Context(), classroomID)
+		if err != nil {
+			return errs.InternalServerError()
+		}
+		notAcceptedWork := numStudents - acceptedWork - startedWork - submittedWork - workInGrading
```

Comment:
If a student were to be on multiple student works at once, this would be incorrect. I don't think we enforce 1 work per student per assignment